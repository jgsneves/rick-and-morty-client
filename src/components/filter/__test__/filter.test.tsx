import { fireEvent, render } from '@testing-library/react';
import Filter, { FilterEpisode } from '..';

describe('Filter test suite', () => {
  const mockedFilterData = {} as FilterEpisode;
  const mockedSetData = jest.fn();
  it('should render proper snapshot', () => {
    const { asFragment } = render(
      <Filter
        label="Filtre por nome"
        filterData={mockedFilterData}
        setData={mockedSetData}
        dataToChange="name"
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('setData should be called once', () => {
    const { getByLabelText } = render(
      <Filter
        label="Filtre por nome"
        filterData={mockedFilterData}
        setData={mockedSetData}
        dataToChange="name"
      />,
    );
    const input = getByLabelText('Filtre por nome');

    fireEvent.change(input, { target: { value: 'casa' } });

    expect(mockedSetData).toBeCalledTimes(1);
    expect(mockedSetData).toBeCalledWith({ name: 'casa' });
  });
});
