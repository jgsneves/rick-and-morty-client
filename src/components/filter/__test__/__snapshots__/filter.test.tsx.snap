// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter test suite should render proper snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-gKXOVf bLqwLA"
  >
    <label
      class="sc-jSMfEi sc-iBkjds geWqUb vFWPx"
      for="name"
    >
      Filtre por nome
    </label>
    <input
      class="sc-ftvSup lqAoO"
      id="name"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
